bio_hansel/quality_check/qc_utils.py

```diff
@@ -0,0 +1,69 @@
+from typing import Tuple
+from bio_hansel.subtype import Subtype
+from pandas import DataFrame
+
+'''
+[get_conflicting_tiles]
+Input: Subtype, DataFrame
+Output: List
+Desc: The purpose of this method is to find positive and negative tiles for the same refposition in the dataframe.
+The method will return a list with the conflicting tiles.
+'''
+
+
+def get_conflicting_tiles(st: Subtype, df: DataFrame) -> list:
+    if st.subtype:
+        if 'is_kmer_freq_okay' in df:
+            dfst = df[(df['subtype'] == str(st.subtype)) & (df['is_kmer_freq_okay'])]
+        else:  # fasta files
+            dfst = df[(df['subtype'] == str(st.subtype))]
+
+        pos_tiles = dfst[dfst['is_pos_tile']]
+        neg_tiles = dfst[dfst['is_pos_tile'] == False]
+        pos_tile_values = '|'.join(pos_tiles['refposition'].values.tolist())
```

Comment:
So it looks like `df['refposition']` is a series of type string since there's entries with `negativeXXXXX`, however, that `negative` info is already extracted into another column `is_pos_tile` so it's redundant and it would make a lot more sense for the `refposition` column to contain integers. That way you don't have to do any string/regex matching hackery and instead just ask if `neg_tiles['refposition'].isin(pos_tile_positions)` where `pos_tile_positions` is a list of ints. Hope that makes sense!

Reply:
Should I change the refposition column to just contain integers then?

Reply:
I think so. These Pandas functions may be useful

- https://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.str.replace.html
- https://pandas.pydata.org/pandas-docs/stable/generated/pandas.to_numeric.html
  - e.g. `df['refposition'] = pd.to_numeric(df['refposition'], downcast='unsigned', errors='coerce')`